src/micro_http/src/request.rs

```diff
@@ -86,24 +89,34 @@ pub struct RequestLine {
 }
 
 impl RequestLine {
-    fn parse_request_line(request_line: &[u8]) -> (&[u8], &[u8], &[u8]) {
+    fn parse_request_line(
+        request_line: &[u8],
+    ) -> std::result::Result<RequestLineParts, RequestError> {
         if let Some(method_end) = find(request_line, &[SP]) {
+            // The slice access is safe because `method_end` is smaller than `request_line` size.
             let method = &request_line[..method_end];
 
-            let uri_and_version = &request_line[(method_end + 1)..];
+            let uri_start = method_end.checked_add(1).ok_or(RequestError::Overflow)?;
+
+            // Slice access is safe because `uri_start` <= `request_line` size.
```

Comment:
Who validates that `uri_start` is <= `request_line`? The above check seem to only check that the integer does not overflow.

Reply:
I added more comments, hope it got a littler clearer.
Also modified `parse_request_line` to return an error if there is no `SP`.